core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3535,6 +3539,17 @@ public void stopSync(AlluxioURI syncPoint)
     return mBlockMaster.getWorkerInfoList();
   }
 
+  /**
+   * Used to augment given context with fsm bound call-trackers
+   * TODO(ggezer): Call this for each call and filter by a new method annotation (@Tracked).
+   *
+   * @param context the operation context
+   */
+  private void addFileSystemMasterCallTrackers(OperationContext context) {
+    // Add state-lock interrupt-cycle call-tracker to given context.
+    context.addCallTracker(mMasterContext.getStateLockManager().getInterruptCycleTracker());
```

Comment:
What is preventing us from just adding the tracker to every single context? I thought it is up to the individual rpc to use it or not?

Reply:
The reasoning was to track tracked-APIs by explicitly working on them.
Currently, if an API decides to use the call-tracker, it must have been explicitly attached otherwise an exception is raised.

I already have a todo in this method to make it general.

Reply:
Since deleteInternal() is used by many, I believe we can't go with fail-if-not-tracked approach anymore. So I might make this method called by all APIs as well.